harvester_e2e_tests/integration/test_networks.py

```diff
@@ -0,0 +1,840 @@
+        check_vm_running(api_client, unique_name, wait_timeout)
+
+        # Check until VM ip address exists
+        check_vm_ip_exists(api_client, unique_name, wait_timeout)
+
+        # get data from running VM and transfer to spec
+        code, data = api_client.vms.get(unique_name)
+        spec = spec.from_dict(data)
+
+        # Switch to vlan network
+        spec.mgmt_network = False
+        spec.add_network("default", "default/" + vlan_network['id'])
+
+        # Update VM spec
+        code, data = api_client.vms.update(unique_name, spec)
+
+        code, data = api_client.vms.restart(unique_name)
+
+        # Check VM start in running state
+        check_vm_running(api_client, unique_name, wait_timeout)
+
+        # Check until VM ip address exists
+        check_vm_ip_exists(api_client, unique_name, wait_timeout)
+
+        # Get VM interface ipAddresses
+        code, data = api_client.vms.vm_instance(unique_name)
+        interfaces_data = data['status']['interfaces']
+        for interface in interfaces_data:
+            ip_addresses = interface['ipAddresses']
+
+        # Check VM start in running state
+        check_vm_running(api_client, unique_name, wait_timeout)
+
+        # Check until VM ip address exists
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+
+        while endtime > datetime.now():
+            code, data = api_client.vms.vm_instance(unique_name)
+            if 'ipAddress' in data['status']['interfaces'][0]:
+                break
+
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Failed to get VM {unique_name} IP address, exceed 10 minutes timed out\n"
+                f"Still got {code} with {data}"
+            )
+
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        ip_addresses = []
+
+        while endtime > datetime.now():
+            code, data = api_client.vms.vm_instance(unique_name)
+            if 'interfaces' in data['status']:
+                interfaces_data = data['status']['interfaces']
+                ip_addresses = []
+                for interface in interfaces_data:
+                    if 'ipAddress' in interface:
+                        ip_addresses.append(interface['ipAddress'])
+
+                if len(ip_addresses) > 0:
+                    if vlan_subnet_prefix in ip_addresses[0]:
+                        break
+                sleep(5)
+        else:
+            raise AssertionError(
+                f"Failed to get VM {unique_name} IP address, exceed 10 minutes\n"
+                f"Still got {code} with {data}"
+            )
+
+        # Ping vlan ip address
+        vlan_ip = ip_addresses[0]
+
+        command = ['/usr/bin/ping', '-c', '50', vlan_ip]
+
+        result = subprocess.check_output(command, shell=False, encoding="utf-8")
+
+        assert result.find("64 bytes from {0}".format(vlan_ip)) > 0, (
+            'Failed to ping VM management IP %s' % (vlan_ip))
```

Comment:
I think the error message is not clear, it didn't point out what the type of NIC using by VM, and from where to do the ping.